src/core/symbology-ng/qgsgraduatedsymbolrendererv2.cpp

```diff
@@ -1386,9 +1399,14 @@ void QgsGraduatedSymbolRendererV2::updateSymbols( QgsSymbolV2 *sym )
   int i = 0;
   foreach ( QgsRendererRangeV2 range, mRanges )
   {
-    QgsSymbolV2 *symbol = sym->clone();
-    symbol->setColor( range.symbol()->color() );
-    updateRangeSymbol( i, symbol );
+    QScopedPointer<QgsSymbolV2> symbol( sym->clone() );
```

Comment:
Exception safety ? I know I'm being a bit stuborn here :)

But indeed it's a matter of principle. RAII is a acquired reflex, I never regretted putting a smart pointer in my code (or a small dedicated resource management class) and I wish I could say the same with raw pointers.